mailpoet/tests/acceptance/Subscriptions/WooCheckoutGuestSubscriptionsCest.php

```diff
@@ -85,7 +85,7 @@ public function checkoutOptInUncheckedNoConfirmation(\AcceptanceTester $i) {
     $customerEmail = 'woo_guest_uncheck@example.com';
     $i->orderProductWithoutRegistration($this->product, $customerEmail, false);
     $i->login();
-    $i->checkSubscriberStatusAndLists($customerEmail, SubscriberEntity::STATUS_UNSUBSCRIBED, null, ['WooCommerce Customers']);
+    $i->checkSubscriberStatusAndLists($customerEmail, SubscriberEntity::STATUS_SUBSCRIBED, null, ['WooCommerce Customers']);
```

Comment:
This is a change that I'm not sure about. Before the code would create the subscriber as `subscribed` in \MailPoet\Segments\WooCommerce::synchronizeGuestCustomer() and then the status would be changed to `unsubscribed` in `\MailPoet\WooCommerce\Subscription::handleSubscriberOptin()` since the optin checkbox is unchecked. Now that we changed the behavior of `handleSubscriberOptin()` and it doesn't unsubscribe subscribers anymore, in the case of this test, we are left with a subscriber with the status `subscribed` but without any list which seems a bit odd, at the same time, I'm not sure if there is a better status.

Reply:
This does seem a bit odd. I'm not sure what the best solution is. One behavioral change is that these users no longer immediately appear when you look at the WooCommerce Customers list. However, if you run the "force sync" operation, they do get added to the list with a status of unsubscribed.

There are other instances where we have inconsistencies between normal subscribing behavior vs running syncs but I wouldn't want to introduce another if we can avoid it. That being said, I think I remember running into this exact issue when I was looking into this flow many months ago and I remember it not seeming like there was a good solution.

Reply:
I'm not sure what is the best solution here. cc @NeosinneR for your input from the product perspective.

Reply:
> What about leaving it unconfirmed (globally) and letting \MailPoet\WooCommerce\Subscription::handleSubscriberOptin() set it to subscribed if the checkbox was checked?

The status is set to `subscribed` when the email confirmation is disabled (I should have mentioned this in the initial comment). So I'm not sure if we want to leave the user with the status `unconfirmed` since email confirmation is disabled. This is what the code currently does:

https://github.com/mailpoet/mailpoet/blob/b80b7d9b1c19156b742354d8cb1a66e412e0bb59/mailpoet/lib/Segments/WooCommerce.php#L165-L168

> One behavioral change is that these users no longer immediately appear when you look at the WooCommerce Customers list. However, if you run the "force sync" operation, they do get added to the list with a status of unsubscribed.

@NeosinneR, what about the behavioral change that @johnolek mentioned? Are we ok with that?

Reply:
> what about the behavioral change that @johnolek mentioned? Are we ok with that?

But in the end, nothing changes functionally, no? Both before and after this PR, you can't contact these contacts, and they are not counted towards limits. The difference is only that they are not immediately seen. If that's the case, I think that's an acceptable change.

Reply:
Yeah, my understanding is that nothing changes in terms of functionality. Thanks for your input.

@johnolek, are you ok if we proceed with this PR the way it is? Please let me know if you prefer that we explore different options.

Reply:
@johnolek, ignore my previous comment. I just saw that you approved the PR. I will check the conflicts.